### DISCUSSION TITLE
DICOM input for GAN

### DISCUSSION THREAD

**Discussion by PCaesar18** [Project-MONAI/MONAI#1672] [Q&A] [answered]
Hi there!

I am working on a project to artificially create CT scans of the parotid gland and was stuck trying to pipeline my own data into the provided GAN tutorial. 

Currently I have a numpy array from an individual CT scan of 512x512 with a depth of 156. (X,Y,Z). Say I would convert this to a similar shaped tensor and add a dimension for the channel. Would this be enough to use as input for my GAN (I also have an X number of full CT scans to make up a batch). Or should I also transform my data into another shape. 

I have my own custom process of loading the required DICOM files and merging them with their corresponding RTstruct files. My concern now is mainly connecting this output and loading it into MONAI's GAN tutorial. 

Looking forward to hearing some recommendations!

**Comment by rijobro** [comment, marked as answer]
Hi, this sounds fine. for 3D data, you want the shape `[C, H, W, D]`. The data loader will take care of creating the batch. You can use `AddChannel` to prepend a singleton channel to your images as part of your transforms.

**Reply by PCaesar18** [reply]
Great! Thanks for your quick answer, i'll give that a shot.
